tools/federation/src/test/java/demos/MonitorRequestsDemo.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package demos;
+
+import org.eclipse.rdf4j.federated.FedXConfig;
+import org.eclipse.rdf4j.federated.FedXFactory;
+import org.eclipse.rdf4j.federated.monitoring.MonitoringUtil;
+import org.eclipse.rdf4j.federated.repository.FedXRepository;
+import org.eclipse.rdf4j.query.QueryLanguage;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+
+public class MonitorRequestsDemo {
+
+	public static void main(String[] args) throws Exception {
+
+		FedXConfig config = new FedXConfig().withEnableMonitoring(true).withLogQueries(true);
+		FedXRepository repo = FedXFactory.newFederation()
+				.withSparqlEndpoint("http://dbpedia.org/sparql")
+				.withSparqlEndpoint("https://query.wikidata.org/sparql")
+				.withConfig(config)
+				.create();
+
+		repo.init();
+
+		String q = "PREFIX wd: <http://www.wikidata.org/entity/> "
+				+ "PREFIX wdt: <http://www.wikidata.org/prop/direct/> "
+				+ "SELECT * WHERE { "
+				+ " ?country a <http://dbpedia.org/class/yago/WikicatMemberStatesOfTheEuropeanUnion> ."
+				+ " ?country <http://www.w3.org/2002/07/owl#sameAs> ?countrySameAs . "
+				+ " ?countrySameAs wdt:P2131 ?gdp ."
+				+ "}";
+
+		try (RepositoryConnection conn = repo.getConnection()) {
+			TupleQuery query = conn.prepareTupleQuery(QueryLanguage.SPARQL, q);
+			try (TupleQueryResult res = query.evaluate()) {
+
+				int count = 0;
+				while (res.hasNext()) {
+					res.next();
+					count++;
+				}
+
+				System.out.println("# Done, " + count + " results");
+			}
```

Comment:
Btw. Counting is nicer with 
```
try(.... stream = Iterations.stream(query.evaluate)){
   .... count = stream.count();
}
```

Reply:
Or, even simpler: 

```
 List<BindingSet> result = QueryResults.asList(query.evaluate());
 System.out.println("# Done, " + result.size() + " results");
```

But it depends on what you're trying to illustrate of course.

Reply:
That keeps all your results in memory.

Reply:
Yes. Typically not a big deal unless you do monstrously big queries - but you're of course correct that streaming for this kind of thing is more efficient. 

I'm fine with either way, also fine if you keep the example as-is @aschwarte10 - the point of this doc is not really to explore the different ways to process RDF4J query results of course :)

Reply:
Thanks for the comments, very useful patterns for the examples, will keep them in mind for future changes

The purpose of this demo is is not the evaluation and consumption of the query, but rather the federation setup and specific configuration options. The system.out.println or in this example the count is just the simplest example. In here we in any case have something like 10 results only.

I would suggest to merge.